tests: net: use zsock APIs instead of POSIX APIs

cfriedt opened the pull request:
Use zsock APIs rather than POSIX APIs in the net subsys testsuites.

It's convention to use native zephyr APIs in the testsuites and POSIX APIs in samples.

cfriedt committed:
tests: net: use zsock APIs instead of POSIX APIs

Use zsock APIs rather than POSIX APIs in net subsys tests.

The primary last remaining API call was `fcntl()` and related
constants, strings, etc.

It's convention to use native zephyr APIs in the testsuites and
POSIX APIs in samples.

Signed-off-by: Chris Friedt <chris@fr4.co>

cfriedt commented:
* removed a number of references to fcntl and posix constants

cfriedt commented on include/zephyr/net/net_compat.h:
```diff
@@ -298,10 +298,6 @@ extern "C" {
-#if !defined(ARPHRD_ETHER)
-#define ARPHRD_ETHER NET_ARPHRD_ETHER
-#endif
-
```
I was getting an warning (promoted to error) about this being redefined elsewhere, and this seemed to fix the issue.

jukkar reviewed (commented):
No objections to the changes, most of those are cosmetic anyway.

cfriedt committed:
net: compat: remove conflicting declaration of ARPHRD_ETHER

Without removing this line, I would get a warning (promoted to error) that
the symbol was being redefined in either `net_ip.h` or
`posix/net/if_arp.h`. Actually, `posix/net/if_arp.h` probably should not
exist, but that is part of a separate PR.

Signed-off-by: Chris Friedt <chris@fr4.co>

cfriedt commented:
@jukkar - should be fixed now. I added the commit after - not sure if you prefer it to be added before. Happy to readjust however you need.
